lib/trento_web/controllers/v1/suse_manager_controller.ex

```diff
@@ -0,0 +1,38 @@
+defmodule TrentoWeb.V1.SUSEManagerController do
+  use TrentoWeb, :controller
+  use OpenApiSpex.ControllerSpecs
+
+  alias Trento.SoftwareUpdates
+
+  alias TrentoWeb.OpenApi.V1.Schema.AvailableSoftwareUpdates.AvailableSoftwareUpdatesResponse
+
+  plug OpenApiSpex.Plug.CastAndValidate, json_render_error_v2: true
+  action_fallback TrentoWeb.FallbackController
+
+  operation :show,
+    summary: "Gets available software updates for a given host",
+    tags: [],
+    description:
+      "Endpoint to fetch available relevant patches and upgradable packages for a given host ID.",
+    parameters: [
+      id: [
+        in: :path,
+        required: true,
+        type: %OpenApiSpex.Schema{type: :string, format: :uuid}
+      ]
+    ],
+    responses: [
+      ok:
+        {"Available software updates for the host", "application/json",
+         AvailableSoftwareUpdatesResponse}
+    ]
+
+  @spec show(Plug.Conn.t(), any) :: Plug.Conn.t()
+  def show(conn, %{id: host_id}) do
+    {:ok, system_id} = SoftwareUpdates.Discovery.get_system_id(host_id)
+    {:ok, relevant_patches} = SoftwareUpdates.Discovery.get_relevant_patches(system_id)
+    {:ok, upgradable_packages} = SoftwareUpdates.Discovery.get_upgradable_packages(system_id)
```

Comment:
@nelsonkopliku can relate. I was actually planning to returning an empty list but this would be very silent in terms of error handling and I wouldn't like that because in the end a user might have 0 upgradable packages when actually behind the scenes it's just a failing call.

So: I like your proposal, but what do you think about an FQDN that still isn't there? Still not found?

Reply:
If we have the host given its id, but that host does not have an fqdn, maybe a 422 could work as well. It'd be consistent with what we often do.

If we return 404 in that case, it'd be great to know whether it is because the host is not in trento or because the host does not have an fqdn.